v2/tools/generator/internal/codegen/pipeline/create_resource_extension_types.go

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/functions"
+)
+
+// CreateResourceExtensionsStageID is the unique identifier of this stage
+const (
+	CreateResourceExtensionsStageID = "createResourceExtensions"
+)
+
+func CreateResourceExtensions(localPath string, idFactory astmodel.IdentifierFactory) Stage {
+	stage := MakeStage(
+		CreateResourceExtensionsStageID,
+		"Create Resource Extensions for each resource type",
+		func(ctx context.Context, state *State) (*State, error) {
+
+			// Map of the new extension types, to all the resource types names on which the extension applies to
+			extendedResourceTypesMapping := make(map[astmodel.TypeName][]astmodel.TypeName)
+			extendedResourceTypes := make(astmodel.Types)
+			resourceTypes := astmodel.FindResourceTypes(state.types)
+
+			// Iterate through resource types and aggregate the resource types that share the same extension in a map.
```

Comment:
When I first read this I thought "extension" meant a common suffix on the type names. 
```suggestion
			// Iterate through resource types and aggregate the resource types that share the same extension type in a map.
```